internal/locate/region_request.go

```diff
@@ -189,8 +190,18 @@ func RecordRegionRequestRuntimeStats(stats map[tikvrpc.CmdType]*RPCRuntimeStats,
 
 // NewRegionRequestSender creates a new sender.
 func NewRegionRequestSender(regionCache *RegionCache, client client.Client) *RegionRequestSender {
+	var v kvrpcpb.APIVersion
+
+	switch regionCache.pdClient.(type) {
```

Comment:
Type assertion looks verbose, we can cache the version somewhere in RegionCache or simply have a method in client to return API version.

Reply:
We could store the API version in the region cache.